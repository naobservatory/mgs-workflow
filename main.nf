include { RUN } from "./workflows/run"
include { RUN2 } from "./workflows/run2"
include { RUN_VALIDATION } from "./workflows/run_validation"
include { INDEX } from "./workflows/index"
include { RUN_DEV_SE } from "./workflows/run_dev_se"

// Configure working and output directories
pubDir  = "${params.base_dir}/output"

workflow {
    if (params.mode == "index") {
        INDEX()
    } else if (params.mode == "run") {
        RUN()
    } else if (params.mode == "run_validation") {
        RUN_VALIDATION()
    } else if (params.mode == "run_dev_se") {
        RUN_DEV_SE()
    }
}

output {
    directory "${pubDir}"
    mode "copy"
}
