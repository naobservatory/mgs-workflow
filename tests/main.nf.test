nextflow_pipeline {

    name "End-to-end test of MGS workflow"
    script "main.nf"

    test("Test index workflow") {
        config "tests/index.config"
        tag "index"

        then {
            assert workflow.success
        }
    }
    test("Test run workflow") {
        config "tests/run.config"
        tag "run"

        then {
            assert workflow.success
        }
    }
    test("Test single-end run workflow") {
        config "tests/run_dev_se.config"
        tag "run_dev_se"

        then {
            assert workflow.success
        }
    }
    test("Test validation workflow") {
        config "tests/run_validation.config"
        tag "validation"

        then {
            assert workflow.success
        }
    }

    test("Test run workflow output") {
        config "tests/run.config"
        tag "run_output"

        then {
            assert workflow.success
            assert snapshot(
                path("${launchDir}/output/results/bracken_reports_merged.tsv.gz"),
                path("${launchDir}/output/results/kraken_reports_merged.tsv.gz"),
                path("${launchDir}/output/results/subset_qc_adapter_stats.tsv.gz"),
                path("${launchDir}/output/results/subset_qc_basic_stats.tsv.gz"),
                path("${launchDir}/output/results/subset_qc_quality_base_stats.tsv.gz"),
                path("${launchDir}/output/results/subset_qc_quality_sequence_stats.tsv.gz"),
                path("${launchDir}/output/results/subset_qc_length_stats.tsv.gz"),
                path("${launchDir}/output/results/virus_clade_counts.tsv.gz"),
                path("${launchDir}/output/results/virus_hits_db.tsv.gz"),
                path("${launchDir}/output/results/blast_hits_paired.tsv.gz"),
                path("${launchDir}/output/results/virus_hits_1.fasta.gz"),
                path("${launchDir}/output/results/virus_hits_2.fasta.gz"),
                path("${launchDir}/output/results/read_counts.tsv.gz"),
            ).match("run_output")
        }

    }
}
