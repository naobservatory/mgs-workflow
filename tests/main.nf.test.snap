{
    "run_output": {
        "content": [
            "bracken_reports_merged.tsv.gz:md5,d116fc06ea955409463686df223ec413",
            "kraken_reports_merged.tsv.gz:md5,5d04e8270bfddc7f435bcc0a76bafbcb",
            "qc_adapter_stats.tsv.gz:md5,5c76643e879a3ad34f18833445c98c1b",
            "qc_basic_stats.tsv.gz:md5,0c3e7dbcf1ed76a6e472ee93e0cc6d38",
            "qc_quality_base_stats.tsv.gz:md5,393490ab4feace88800d7eaf8760451f",
            "qc_quality_sequence_stats.tsv.gz:md5,fdfd0a4e2ff7514f9828ad141878ebc6",
            "virus_clade_counts.tsv.gz:md5,89f68c2cc7912f37b22745168b2fc7e4",
            "virus_hits_db.tsv.gz:md5,72e692fb3d168dc5199ab97ac4c58426",
            "blast_hits_paired.tsv.gz:md5,6dce9cbf1aa6397a9145c9752aa4c723",
            "virus_hits_1.fasta.gz:md5,941487995f144f88df4c2994f2fc551c",
            "virus_hits_2.fasta.gz:md5,e285ccfb350fed7f0e73e29aaa1f10c1"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-13T22:37:38.227517631"
    }
}